sw/host/ot_certs/src/template/mod.rs

```diff
@@ -77,13 +77,19 @@ pub struct Certificate {
     pub basic_constraints: Option<BasicConstraints>,
     /// X509 Subject Alternative Name extension, optional.
     #[serde(default)]
-    pub subject_alt_name: IndexMap<AttributeType, Value<String>>,
+    pub subject_alt_name: Name,
     /// Non-standard X509 certificate extensions.
-    pub extensions: Vec<CertificateExtension>,
+    #[serde(default)]
+    pub private_extensions: Vec<CertificateExtension>,
     /// X509 certificate's signature.
     pub signature: Signature,
 }
 
+/// An X501's Name: this is an ordered list of sets of attributes.
+/// The order of the list matters but the order within the sets of
+/// attributes is arbitrary.
+pub type Name = Vec<IndexMap<AttributeType, Value<String>>>;
```

Comment:
Just curious, why is this a Vec of IndexMaps? Doesn't the IndexMap itself  give you ordering: https://docs.rs/indexmap/latest/indexmap/map/struct.IndexMap.html#order ?

Also, if you do need the Vec, do the IndexMap's only ever have on key/value pair? If so would it be better to use a Vec of Structs or Vec of Tuples here?

Reply:
I have updated the comment, is that better?

Reply:
LGTM, thanks!